src/tstester/base.ts

```diff
@@ -0,0 +1,195 @@
+// Contains foundational abstractions for the testing system
+import { JDBus } from "../jdom/bus"
+import { assert } from "../jdom/utils"
+import { ConsoleUi } from "./jacdac-tstester"
```

Comment:
use an interface here so that you're not tie to your particular impl of ConsoleUi. If we want to use this in React, it'll be implemented differently.

Reply:
Done.